tests/unit/bokeh/core/property/test_container.py

```diff
@@ -144,6 +144,29 @@ def test_init(self) -> None:
         with pytest.raises(TypeError):
             bcpc.Dict()
 
+    def test_validate_invalid_key(self) -> None:
+        d = bcpc.Dict(String, String)
+        with pytest.raises(ValueError) as err:
+            d.validate({"Foo": "Bar", 1: "Baz"})
+            assert("invalid keys: 1" in err.value.args[0])
```

Comment:
No need for parens with asserts (here and below):
```suggestion
            assert "invalid keys: 1" in err.value.args[0]
```
It might be worth seeing if there is a simpler express, e.g. would ` "invalid keys: 1" in str(err)` work? (not sure, just a guess)

Reply:
`str(err) == "<ExceptionInfo ValueError('expected a dict of type Dict(String, String), got a dict with invalid keys: 1') tblen=2>"`

`str(err.value) == "expected a dict of type Dict(String, String), got a dict with invalid keys: 1"`

I'm going with the second option unless if you want otherwise

Reply:
👍 sounds good!